modules/iam-role-for-service-accounts-eks/policies.tf

```diff
@@ -531,24 +531,23 @@ data "aws_iam_policy_document" "karpenter_controller" {
       "ec2:DescribeInstanceTypeOfferings",
       "ec2:DescribeAvailabilityZones",
       "ec2:DescribeSpotPriceHistory",
+      "ec2:RunInstances",
+      "iam:PassRole",
       "pricing:GetProducts",
+      "ssm:GetParameter",
     ]
-
-    resources = ["*"]
   }
 
   statement {
-    actions = [
-      "ec2:TerminateInstances",
-      "ec2:DeleteLaunchTemplate",
-    ]
-
+    sid       = "ConditionalEC2Termination"
+    effect    = "Allow"
     resources = ["*"]
+    actions   = ["ec2:TerminateInstances"]
 
     condition {
-      test     = "StringEquals"
-      variable = "ec2:ResourceTag/${var.karpenter_tag_key}"
-      values   = [var.karpenter_controller_cluster_id]
+      test     = "StringLike"
+      variable = "ec2:ResourceTag/Name"
```

Comment:
this makes some strong assumptions that go against feedback from users and is also a breaking change

Reply:
What feedback did you get about it ? This change seems to work well for us

Reply:
Newer versions of karpenter have a new label that can make this delete policy depend on it, instead of the name like it did in the past.
Hard coding it back to just name is bad.